quickstarts/08-data-agent-mcp-chainlit/README.md

```diff
@@ -0,0 +1,146 @@
+
+## Key Benefits
+
+- **Conversational Knowledge Base**: Users can talk to their database in natural language, ask complex questions and perform advanced analysis over data
+- **Conversational Memory**: The agent maintains context across interactions in the user's [database of choice](https://docs.dapr.io/reference/components-reference/supported-state-stores/)
+- **UI Interface**: Use an out-of-the-box, LLM-ready chat interface using [Chainlit](https://github.com/Chainlit/chainlit)
+- **Boilerplate-Free DB Layer**: MCP allows the Dapr Agent to connect to the database without requiring users to write Postgres-specific code
+
+## Prerequisites
+
+- Python 3.10 (recommended)
+- pip package manager
+- OpenAI API key (for the OpenAI example)
+- [Dapr CLI installed](https://docs.dapr.io/getting-started/install-dapr-cli/)
+
+## Environment Setup
+
+```bash
+# Create a virtual environment
+python3.10 -m venv .venv
+
+# Activate the virtual environment 
+# On Windows:
+.venv\Scripts\activate
+# On macOS/Linux:
+source .venv/bin/activate
+
+# Install dependencies
+pip install -r requirements.txt
+
+# Initialize Dapr
+dapr init
+```
+
+## LLM Configuration
+
+For this example, we'll be using the OpenAI client that is used by default. To target different LLMs, see [this example](../02_llm_call_dapr/README.md).
+
+Create a `.env` file in the project root:
+
+```env
+OPENAI_API_KEY=your_api_key_here
+```
+
+Replace `your_api_key_here` with your actual OpenAI API key.
+
+## Postgres Configuration
+
+### Connect to an existing database
+
+Create an `.env` file in the root directory of this quickstart and insert your database configuration:
+
+```bash
+DB_HOST=<HOST>
+DB_PORT=<PORT>
+DB_NAME=<DATABASE-NAME>
+DB_USER=<USER>
+DB_PASSWORD=<PASSWORD>
+```
+
+### Create a new sample database
+
+First, install Postgres on your machine.
+
+#### Option 1: Using Docker
+
+```bash
+docker run --name sampledb -e POSTGRES_PASSWORD=mypassword -e POSTGRES_USER=admin -e POSTGRES_DB=userdb -p 5432:5432 -d postgres`
+```
+
+#### Option 2: Using Brew
+
+```bash
+brew install postgresql
+brew services start postgresql
+
+psql postgres
+> CREATE USER admin WITH PASSWORD 'mypassword';
+> CREATE DATABASE userdb 
+> GRANT ALL PRIVILEGES ON DATABASE userdb TO admin;
+```
+
+#### Create the table and data
+
+Next, create the users table and seed data:
+
+```bash
+psql -h localhost -U admin -d userdb -f schema.sql
+psql -h localhost -U admin -d userdb -f users.sql
+```
+
+#### Create .env file
+
+Finally, create an `.env` file in the root directory of this quickstart and insert your database configuration:
+
+```bash
+DB_HOST=localhost
+DB_PORT=5432
+DB_NAME=userdb
+DB_USER=admin
+DB_PASSWORD=mypassword
+```
+
+## MCP Configuration
+
+To get the Dapr Agent to connect to our Postgres database, we'll use a Postgres MCP server.
+Change the settings below based on your Postgres configuration:
```

Comment:
Can we say, "By default, if you are running your postgresql DB locally (i.e. in a docker container), the `<HOST>` value should be `localhost` in here `DATABASE_URI=postgresql://<USERNAME>:<PASSWORD>@<HOST>:5432/userdb`

Reply:
Done